packages/typescript-estree/src/convert.ts

```diff
@@ -1585,6 +1585,8 @@ export class Converter {
               hasModifier(SyntaxKind.ReadonlyKeyword, node) || undefined,
             static: hasModifier(SyntaxKind.StaticKeyword, node) || undefined,
             export: hasModifier(SyntaxKind.ExportKeyword, node) || undefined,
+            override:
+              hasModifier(SyntaxKind.OverrideKeyword, node) || undefined,
```

Comment:
I do hate that this emits `"override": undefined` if there isn't an override - but the entire node is built like this so 🤷‍♂️  it'll do for now.

We can revisit the entire AST later to fix this sort of weirdness and standardise on just emitting `false`.